tests/unit/migrations/migrationUtils.test.ts

```diff
@@ -0,0 +1,155 @@
+
+    expect(filteredMigrations).toStrictEqual([migrations[0]]);
+  });
+});
+
+describe("getMigrationsToSkip", () => {
+  it("Correctly return skipped sucesfully migrated migrations", () => {
+    const enviromnentStatus: ReadonlyArray<MigrationStatus> = [
+      createMigrationStatus(migrations[0].name, migrations[0].module.order, true, "run"),
+      createMigrationStatus(migrations[2].name, migrations[2].module.order, true, "run"),
+    ];
+
+    const skippedMigrations = getMigrationsToSkip(enviromnentStatus, migrations, "run");
+
+    expect(skippedMigrations).toStrictEqual([migrations[0], migrations[2]]);
+  });
+
+  it("Correctly return skipped sucesfully migrated migrations and does not return not sucessfull migration", () => {
+    const enviromnentStatus: ReadonlyArray<MigrationStatus> = [
+      createMigrationStatus(migrations[0].name, migrations[0].module.order, true, "run"),
+      createMigrationStatus(migrations[2].name, migrations[2].module.order, false, "run"),
+    ];
+    const skippedMigrations = getMigrationsToSkip(enviromnentStatus, migrations, "run");
+
+    expect(skippedMigrations).toStrictEqual([migrations[0]]);
+  });
+
+  it("Correctly return does not return any migration when operation is oposite", () => {
+    const enviromnentStatus: ReadonlyArray<MigrationStatus> = [
+      createMigrationStatus(migrations[0].name, migrations[0].module.order, true, "run"),
+      createMigrationStatus(migrations[1].name, migrations[1].module.order, false, "run"),
+    ];
+    const skippedMigrations = getMigrationsToSkip(enviromnentStatus, migrations, "rollback");
+
+    expect(skippedMigrations).toStrictEqual([migrations[2], migrations[3], migrations[4]]);
+  });
+});
+
+describe("getMigrationsWithDuplicateOrders", () => {
+  it.each([
+    [[{ ...migrations[0] }, { ...migrations[0] }]],
+    [[{ ...migrations[2] }, { ...migrations[2] }]],
+  ])("Correctly return non empty map when multiple migrations have same order", (c) => {
+    const sameOrderMigrations = getMigrationsWithDuplicateOrders(c);
+    expect(sameOrderMigrations).toStrictEqual(new Map([[c[0]?.module.order, [c[0], c[1]]]]));
+  });
+});
+
+describe("executeMigrations", () => {
+  it("Correctly executesMigration", async () => {
+    const migrationsToRun = [migrations[0], migrations[1]];
+
+    const migrationStatus = await executeMigrations(
+      migrationsToRun,
+      new ManagementClient({ environmentId: "-", apiKey: "-" }),
+      { continueOnError: false, operation: "run" },
+      { logLevel: "none" },
+    );
+
+    const migrationStatusWithoutDate = migrationStatus.status.map(s => ({ ...s, time: undefined }));
+    const expectedResult = [
+      createMigrationStatus(migrations[0].name, migrations[0].module.order, true, "run"),
+      createMigrationStatus(migrations[1].name, migrations[1].module.order, true, "run"),
+    ].map(s => ({ ...s, time: undefined }));
+
+    expect(migrationStatusWithoutDate).toStrictEqual(expectedResult);
```

Comment:
I would be nice to check that the `run` functions actually got called with a properly setup client. This is, after all, the most important part of the whole migration command. 😄